src/services/ui/src/components/UsaBanner/index.tsx

```diff
@@ -4,10 +4,14 @@ import { LockIcon } from "../LockIcon";
 import { GovernmentBuildingIcon } from "../GovernmentBuildingIcon";
 import UsFlag from "@/assets/us_flag_small.png";
 import { useMediaQuery } from "@/hooks";
+import { useUserContext } from "../Context/userContext";
 
 export const UsaBanner = () => {
   const [isOpen, setIsOpen] = useState(false);
   const isDesktop = useMediaQuery("(min-width: 640px)");
+  const userContext = useUserContext();
+  const role = userContext?.user?.["custom:cms-roles"]?false:true;
```

Comment:
Since `userContext` is technically returning a React State item, role would be safer with `useMemo`. React State is fickle in my experience, so hooks like `useMemo` allow us to derive values from a piece of state and update that value when the state changes. We can also simplify your later check by including the `userContext?.user` boolean check from line 66 in the memoized variable.

For example:
```typescript
const role = userContext?.user?.["custom:cms-roles"] ? false : true;
// ...
{ role && userContext?.user && ( ... )}
```
can be memoized and used as such
```typescript
const hasAccess = useMemo(() =>  
  // checks both user data is present and user has cms-roles
  userContext?.user && userContext?.user?.["custom:cms-roles"],
  [userContext] // this is the "dependency array", any React State used in memos are dependencies
)
// ...
{ hasAccess && ( ... )}
```

Reply:
[useMemo React docs](https://react.dev/reference/react/useMemo)